src/main/scala/workflow/graph/GatherTransformer.scala

```diff
@@ -0,0 +1,19 @@
+package workflow.graph
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * A [[TransformerOperator]] that gathers multiple datasets of {@tparam T} into a dataset of Seq[T]
+ * (Or individual datums of T into a single Seq[T])
+ */
+private[workflow] case class GatherTransformer[T]() extends TransformerOperator {
+  override private[graph] def singleTransform(inputs: Seq[DatumExpression]): Any = {
+    inputs.map(_.get.asInstanceOf[T])
+  }
+
+  override private[graph] def batchTransform(inputs: Seq[DatasetExpression]): RDD[_] = {
```

Comment:
Same as above. The methods in TransformerOperator we're overriding are untyped, so returning `RDD[_]` is fine, but if we wanted to we could type it as `RDD[Seq[T]]` because  `RDD[Seq[T]] <: RDD[_]`